cpp/include/cuml/tree/decisiontree.hpp

```diff
@@ -145,23 +145,32 @@ struct TreeMetaDataNode {
 };
 
 /**
- * @brief Print high-level tree information.
+ * @brief Obtain high-level tree information.
  * @tparam T: data type for input data (float or double).
  * @tparam L: data type for labels (int type for classification, T type for regression).
  * @param[in] tree: CPU pointer to TreeMetaDataNode
+ * @return High-level tree information as string
  */
 template <class T, class L>
-void print_tree_summary(const TreeMetaDataNode<T, L> *tree);
+std::string print_tree_summary(const TreeMetaDataNode<T, L> *tree);
```

Comment:
should we rename from print_tree_summary to string_tree_summary or something similar?

Reply:
Yes, let's rename it, since we are making a breaking change to the function signature.

Reply:
I've renamed `print_rf_*` functions to `get_rf_*_text`.